didcomm_messaging/v1/crypto/nacl.py

```diff
@@ -0,0 +1,222 @@
+"""V1CryptoService implementation for pynacl."""
+
+from dataclasses import dataclass
+from typing import Dict, Optional, OrderedDict, Sequence, Tuple
+
+import base58
+from pydid import VerificationMethod
+
+from didcomm_messaging.crypto.base import PublicKey, SecretKey, SecretsManager
+from didcomm_messaging.crypto.jwe import JweBuilder, JweEnvelope, JweRecipient
+from didcomm_messaging.multiformats import multibase, multicodec
+
+from .base import V1CryptoService, V1UnpackResult, RecipData
+
+try:
+    import nacl.bindings
+    import nacl.exceptions
+    import nacl.utils
+except ImportError as err:
+    raise ImportError(
+        "V1 nacl implementation requires 'nacl' extra to be installed"
+    ) from err
+
+
+@dataclass
+class KeyPair(SecretKey):
+    """Keys."""
+
+    verkey: bytes
+    sigkey: bytes
+
+    @property
+    def verkey_b58(self) -> str:
+        """Return base58 encoding of verkey."""
+        return base58.b58encode(self.verkey).decode()
+
+    @property
+    def kid(self) -> str:
+        """Get the key ID."""
+        return self.verkey_b58
+
+
+@dataclass
+class EdPublicKey(PublicKey):
+    """Simple public key representation as base58 encoded str."""
+
+    value: bytes
+
+    @classmethod
+    def from_verification_method(cls, vm: VerificationMethod) -> "EdPublicKey":
+        """Create a Key instance from a DID Document Verification Method."""
+        key_bytes = cls.key_bytes_from_verification_method(vm)
+        return EdPublicKey(key_bytes)
+
+    @property
+    def key(self) -> str:
+        """Return base58 encoded key."""
+        return base58.b58encode(self.value).decode()
+
+    @property
+    def kid(self) -> str:
+        """Get the key ID."""
+        return self.key
```

Comment:
Would this hold true for the kid if the did was `did:peer:1/2` during a DIDComm V1 exchange?